app/uk/gov/hmrc/cgtpropertydisposalsfrontend/connectors/AddressLookupConnector.scala

```diff
@@ -45,33 +48,31 @@ class AddressLookupConnectorImpl @Inject() (
   servicesConfig: ServicesConfig
 )(implicit
   ec: ExecutionContext
-) extends AddressLookupConnector {
+) extends AddressLookupConnector
+    with Logging {
 
-  val url: String =
-    servicesConfig.baseUrl("address-lookup") + "/lookup"
+  val url: String = servicesConfig.baseUrl("address-lookup") + "/lookup"
 
   val headers: Seq[(String, String)] = {
-    val userAgent = servicesConfig.getString(
-      "microservice.services.address-lookup.user-agent"
-    )
+    val userAgent = servicesConfig.getString("microservice.services.address-lookup.user-agent")
     Seq("User-Agent" -> userAgent)
   }
 
   override def lookupAddress(postcode: Postcode, filter: Option[String])(implicit
     hc: HeaderCarrier
-  ): EitherT[Future, Error, HttpResponse] = {
-
-    val lookupAddressByPostcode = LookupAddressByPostcode(
+  ): EitherT[Future, Error, AddressLookupResponse] = {
+    val body = LookupAddressByPostcode(
       postcode.value.replace(" ", "").toUpperCase(Locale.UK),
       filter.map(f => URLEncoder.encode(f, "UTF-8"))
     )
-
-    EitherT[Future, Error, HttpResponse](
-      http
-        .POST[LookupAddressByPostcode, HttpResponse](url, lookupAddressByPostcode, headers)
-        .map(Right(_))
-        .recover { case e => Left(Error(e)) }
-    )
+    http
+      .POST[LookupAddressByPostcode, Either[UpstreamErrorResponse, AddressLookupResponse]](url, body, headers)
+      .map(_.left.map { error =>
+        logger.error(s"POST to $url failed", error)
+        Error(s"Response to address lookup came back with status ${error.statusCode}")
```

Comment:
Can we use  'UpstreamErrorResponse' instead of 'Response'

Reply:
We can, but it doesn't make it any more accurate. The UpstreamErrorResponse is an object representing a Response. UpstreamErrorResponses aren't sent by downstream, Responses are. Also, this is a comment about the other PR, which has been merged since.